volttrontesting/platform/dbutils/test_mysqlfuncts.py

```diff
@@ -0,0 +1,606 @@
+                        value_string TEXT NOT NULL, topics_list TEXT, 
+                        UNIQUE(topic_id, ts), INDEX (ts ASC))
+                    """
+            seed_database(container, query)
+
+            agg_topic_id = 42
+            agg_type = "AVG"
+            period = "1776"
+            ts = "2020-06-01 12:30:59"
+            data = "some_data"
+            topic_ids = [12, 54, 65]
+            expected_data = (datetime.datetime(2020, 6, 1, 12, 30, 59), 42, '"some_data"', '[12, 54, 65]')
+
+            res = mysqlfuncts.insert_aggregate(agg_topic_id, agg_type, period, ts, data, topic_ids)
+
+            assert res is True
+            assert get_data_in_table(port_on_host, "AVG_1776")[0] == expected_data
+
+
+@pytest.mark.mysqlfuncts
+def test_collect_aggregate_should_raise_value_error(get_container_func, ports_config):
+    get_container, image = get_container_func
+    with get_container(image, ports=ports_config["ports"], env=ENV_MYSQL) as container:
+        wait_for_connection(container)
+        create_all_tables(container)
+
+        port_on_host = ports_config["port_on_host"]
+        with get_mysqlfuncts(port_on_host) as mysqlfuncts:
+            query = f"""
+                        REPLACE INTO {DATA_TABLE}
+                        VALUES ('2020-06-01 12:30:59', 42, '2');
+                        REPLACE INTO {DATA_TABLE}
+                        VALUES ('2020-06-01 12:31:59', 43, '8')
+                    """
+            seed_database(container, query)
+
+            topic_ids = [42, 43]
+            agg_type = "avg"
+            expected_aggregate = (5.0, 2)
+
+            actual_aggregate = mysqlfuncts.collect_aggregate(topic_ids, agg_type)
+
+            assert actual_aggregate == expected_aggregate
+
+
+@pytest.mark.mysqlfuncts
+def test_collect_aggregate_should_raise_value_error(get_container_func, ports_config):
+    get_container, image = get_container_func
+    with get_container(image, ports=ports_config["ports"], env=ENV_MYSQL) as container:
+        wait_for_connection(container)
+
+        with pytest.raises(ValueError):
+            with get_mysqlfuncts(ports_config["port_on_host"]) as mysqlfuncts:
+                mysqlfuncts.collect_aggregate("dfd", "Invalid agg type")
+
+
+@contextlib.contextmanager
+def get_mysqlfuncts(port):
+    connect_params = {"host": 'localhost',
+                      "port": port,
+                      "database": TEST_DATABASE,
+                      "user": "root",
+                      "passwd": ROOT_PASSWORD}
+
+    table_names = {"data_table": DATA_TABLE,
+                   "topics_table": TOPICS_TABLE,
+                   "meta_table": META_TABLE,
+                   "agg_topics_table": AGG_TOPICS_TABLE,
+                   "agg_meta_table": AGG_META_TABLE}
+
+    mysqlfuncts = MySqlFuncts(connect_params, table_names)
+
+    yield mysqlfuncts
+
+
+@pytest.fixture(params=IMAGES)
+def get_container_func(request):
```

Comment:
This fixture gives the capability to test multiple versions of MySQL. It returns both the docker_wrapper function and a list of images that will be used to create a container. In other words, it acts like a for-loop for each test so that it can be run on different versions of MySQL.